### ISSUE TITLE
Enhancement: Use title of linked page as hyperlink description

### ISSUE THREAD

**Issue by RohanOne** [Laverna/laverna#624]
when inserting a hyperlink into a page, a message comes up stating "enter link description here". could it be possible to place the title of the page instead, as the description? More likely than not, the title of the linked page is what the user really wants. And if this is done, is it possible to still show the user in some way (maybe a tooltip) that the text is actually the link description and can be edited to provide an alternative text?

**Comment by JerJohn15** [comment]
@RohanOne 

> **when inserting a hyperlink into a page, a message comes up stating "enter link description here"**. could it be possible to place the title of the page instead, as the description? More likely than not, the title of the linked page is what the user really wants. And if this is done, is it possible to still show the user in some way (maybe a tooltip) that the text is actually the link description and can be edited to provide an alternative text?

So are you asking if it is possible to extract text from the web address field inside the user's web browser (Firefox, Chrome) and a hyperlink that is show by way of text (like a title of a news article on CNN.com), and include it as the description of the link instead of it's default title: "link" (or having to let the user manually enter in the title itself)? So essentially your automating the process of having to enter in a description for a hyperlink manually. For example:

**A link from the browser's URL input field***

www.github.com 

becomes

[GitHub](https://www.github.com)

when extracted.

**A hyperlink shown as text**

"Flying? Don't use Galaxy Note 7"  <--- taken from the front page of cnn.com

becomes 

[Flying? Don't use Galaxy Note 7](http://money.cnn.com/2016/09/08/technology/samsung-galaxy-note-7-airlines-battery-fires/index.html)

when extracted.

If this is what your asking, I think it's certainly possible. However, one would have to also take into consideration what happens if the user wants to enter in a link for an image or thumbnail icon (GitHub avatar's for instance), or even embedded videos that may not contain text that refers back to a link, if we even want to implement a feature this far.

**Comment by RohanOne** [comment]
yeah, this is what i meant, along with the ability to link to other notes within laverna and get the title of the linked note autofilled

**Comment by memeplex** [comment]
The title of the note is already entered by the user in the add link dialog so it should be easy to provide the autofill in this simple wiki-like style at least.
